lib/module.js

```diff
@@ -219,17 +219,25 @@ if (process.platform === 'win32') {
     // note: this approach *only* works when the path is guaranteed
     // to be absolute.  Doing a fully-edge-case-correct path.split
     // that works on both Windows and Posix is non-trivial.
+
+    // return root node_modules when path is 'D:\\'.
+    // path.resolve will make sure from.length >=3 in Windows.
+    if (from[from.length - 1] === '\\' && from[from.length - 2] === ':')
```

Comment:
Actually on second thought I think it is probably safe to assume `from.length >= 2` on Windows. So just the `charCodeAt()` changes should be sufficient.